### DISCUSSION TITLE
Getting a promise back from hooks

### DISCUSSION THREAD

**Discussion by Koowah** [wevm/wagmi#1581] [Q&A] [answered]
Hi guys,

As the title says, I was wondering if it was possible to get the `promise` from `useContractWrite` for instance (to track its resolution) ? I know we can **indirectly** access it through `isLoading` and `isSuccess` along with a `useEffect` (any other ideas ?), but for my specific use case (would like to use await for successive contract calls), the promise would be of great help.

Thanks for your assistance
Cheers😄

**Comment by SaadNiaziDev** [comment, marked as answer]
Did you used writeAsync?
i guess it would help you
writeAsync: ((args?: WriteContractConfig) => Promise<TransactionResponse>) | undefined

There is another workaround as well,
wrap write function in Promise() and  resolve useContractWrite on onSuccess or onError event.

**Reply by Koowah** [reply]
No I didn't see writeAsync. I will try it. I like the idea of wrapping the code in Promise().
So if I wanted to `await` a contract call I would have to do something like this :
```
const {config} = usePrepareContractWrite ...
const {data, writeAsync} = useContractWrite ...
const {isLoading, isSuccess} = useWaitForTransaction({hash: data?.hash, confirmations: 2})
await writeAsync()
```
Or like this :
```
const {config} = usePrepareContractWrite ...
const {data, write} = useContractWrite ...
const {isLoading, isSuccess} = useWaitForTransaction({hash: data?.hash, confirmations: 2})
const examplePromise = new Promise((resolve, reject) => {
write()
if (isSuccess) resolve
}
await examplePromise()
```

Is that alright ? Do you see any way to improve it ?

**Reply by SaadNiaziDev** [reply]
Method #1 is simple and it should get work done  :P
